### DISCUSSION TITLE
Shopify Functions - Only run if customer is logged in

### DISCUSSION THREAD

**Discussion by ricksnathan** [Shopify/function-examples#250] [Feedback: Discounts APIs]
Hi all,

I've created a product discount that's based on a customer's tags, i.e. "Employee". Whenever I reach checkout, anybody that knows that email can use it and receive the discount, since there is no logged in state of the customer. They'll get the discount, but all order information will go to the wrong email (i.e. the person who is an actual employee).  What checks are possible to make sure the user is logged in? I'd rather not have customers log in before checkout! 

- [Reference](https://stackoverflow.com/questions/75929567/limit-shopify-function-to-logged-in-customers-only/76368521):

**Comment by davejcameron** [comment]
This should now be supported in `2023-07` where you can look at https://shopify.dev/docs/api/functions/reference/product-discounts/graphql/common-objects/buyeridentity `isAuthenticated` property.

**Comment by ricksnathan** [comment]
Cheers!
